flips/20211007-transaction-fees.md

```diff
@@ -0,0 +1,264 @@
+
+- <img src="https://render.githubusercontent.com/render/math?math=F_i">: Inclusion fees; Represent the cost of including the transaction in a collection.
+    - <img src="https://render.githubusercontent.com/render/math?math=f">: The current cost of one unit of effort.
+    - <img src="https://render.githubusercontent.com/render/math?math=E_i">: Inclusion effort; The time invariant effort required to handle this transaction (include it into a collection, check signatures are valid, check that the payer can pay, transfer the transaction over the wire,...). The inclusion effort is known before the transaction is executed.
+    - <img src="https://render.githubusercontent.com/render/math?math=E_c">: Execution effort; The time invariant effort required to execute this transaction (run the script, process the results, send results to verification node, verify the results, ...). The Execution effort cannot be know without executing the transactions script.
+
+
+### Fee effort cost factor
+
+The effort cost factor <img src="https://render.githubusercontent.com/render/math?math=f"> should be defined on a smart contract and adjustable via the service account admin resource and should be accessible for everyone to read. 
+
+The effort cost factor can be used in three different ways:
+
+1. The main usage of the effort cost factors is adjusting it when the network is under load. 
+
+2. The second reason to tweak the effort cost factors is that over time as hardware improves, the same amount of effort will be easier to handle by the nodes.
+
+3. A potential third use of the effort cost factor would be to keep the cost of transaction stable according to the USD value of flow. This would require a periodic job to run, check the USD value of FLOW and update the cost factor to account for any changes. This would allow for the average transaction fees to stay relatively close to a set USD value.
+
+While the effort cost factor could be adjusted manually from the start, that would mean responding to network load would be very slow. A potential solution to adjusting the effort cost factor automatically would be a service that monitored the networks performance and sent a transaction to update the fee factors when there was a change in the network load.
+
+### Inclusion effort
+
+Inclusion effort can represent many different parts of the system. Listed below are some potential factors that may play into the inclusion effort part of the fees listed by node type. Adding factors to this list (and actually implementing them) can happen gradually. One way to notice additional factors that might play into the inclusion effort is to compare transactions that have similar inclusion effort but cause a different load on the system.
+
+1. Access node:
+    1. Sending a transaction causes load on the access node (might depend on the byte size of the transaction).
+    2. The access node needs to potentially do a check (at a constant effort).
```

Comment:
"The access node needs to potentially do a check (at a constant effort)" this statement is incomplete. My focus probably jumped here. The statement is supposed to be: "The access node needs to potentially do a check that the payer can pay (at a constant effort)".

As for the access node checking for the syntactical correctness of the cadence code, I have some thoughts:
- I checked and currently the access node does check that the script is syntactically correct, so this should be added to the list of things the access node needs to do, but the cost of this might be difficult to determine, without actually checking for syntactical correctness.
- What motivates the access node to actually do this? Will it be slashed if it submits an invalid transaction?
- The execution node has to re-perform this check anyway otherwise it cannot execute it, so this cost probably needs to be counted twice.

I will add it to this list so we can look into how costly checking for syntactical correctness and if it needs to be included in the effort calculation sooner or later.